### ISSUE TITLE
dart 2.9 dart2native fails due to use of --packages

### ISSUE THREAD

**Issue by bsutton** [dart-lang/sdk#42976]
I've just upgraded to dart 2.9.

When I try to compile a script the compile fails with:
```
Scheme not starting with alphabetic character (at character 1)
```
It appears that the problem is caused by my using the --package switch.
If I remove the --package switch the compile completes as normal.

The error points to different packages depending on whether I launch with an absolute or a relative path.

```
dart2native bin/install.dart --packages=../.dart_tool/package_config.json --output=bin/install

Unhandled exception:
FormatException: Scheme not starting with alphabetic character (at character 1)
 "file:///home/bsutton/.pub-cache/hosted/pub.dartlang.org/archive-2.0.13",
^

#0      _Uri._fail (dart:core/uri.dart:1628:5)
#1      _Uri._makeScheme (dart:core/uri.dart:2167:7)
#2      new _Uri.notSimple (dart:core/uri.dart:1459:18)
#3      Uri.parse (dart:core/uri.dart:1022:17)
#4      _Uri.resolve (dart:core/uri.dart:2592:27)
#5      convertToPackageUri (package:vm/kernel_front_end.dart:697:39)
<asynchronous suspension>
#6      runCompiler (package:vm/kernel_front_end.dart:240:21)
#7      compile (file:///tmp/tmp2mIS3S/dart-2.9.0/dart/pkg/vm/bin/gen_kernel.dart:39:10)
#8      main (file:///tmp/tmp2mIS3S/dart-2.9.0/dart/pkg/vm/bin/gen_kernel.dart:34:25)
#9      _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:299:32)
#10     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:168:12)

Failed to generate native files:
Generating AOT kernel dill failed!
```

and
```
/usr/lib/dart/bin/dart2native /home/bsutton/git/auditor/dart_tools/bin/install.dart --packages=/home/bsutton/git/auditor/dart_tools/.dart_tool/package_config.json --output=/home/bsutton/git/auditor/dart_tools/bin/install

Unhandled exception:
FormatException: Scheme not starting with alphabetic character (at character 1)
 "file:///home/bsutton/.pub-cache/hosted/pub.dartlang.org/_fe_analyzer_shar...
^

#0      _Uri._fail (dart:core/uri.dart:1628:5)
#1      _Uri._makeScheme (dart:core/uri.dart:2167:7)
#2      new _Uri.notSimple (dart:core/uri.dart:1459:18)
#3      Uri.parse (dart:core/uri.dart:1022:17)
#4      _Uri.resolve (dart:core/uri.dart:2592:27)
#5      convertToPackageUri (package:vm/kernel_front_end.dart:697:39)
<asynchronous suspension>
#6      runCompiler (package:vm/kernel_front_end.dart:240:21)
#7      compile (file:///tmp/tmp2mIS3S/dart-2.9.0/dart/pkg/vm/bin/gen_kernel.dart:39:10)
#8      main (file:///tmp/tmp2mIS3S/dart-2.9.0/dart/pkg/vm/bin/gen_kernel.dart:34:25)
#9      _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:299:32)
#10     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:168:12)

Failed to generate native files:
Generating AOT kernel dill failed!
```

**Comment by mraleph** [comment]
Note: you are giving it a package config file, not packages file. packages file is called `.packages`, hence the error.

**Comment by bsutton** [comment]
Not according to the spec:

From: https://github.com/dart-lang/language/blob/master/accepted/future-releases/language-versioning/package-config-file-v2.md

```
Specifying the File
Tools currently allow you to specify the .packages file with a command line argument:

--packages=someFile
-p someFile
This approach will work with the JSON configuration file as well. 
```

**Comment by mraleph** [comment]
You are right, my bad (fwiw the command line help seems to be outdated for some of the tools). 

There was a bug https://github.com/dart-lang/sdk/issues/42614 which is now fixed on master, but not on 2.9.

**Comment by bsutton** [comment]
No problem.

As long as a fix is on its way iny happy.
